Core/Src/monitor.c

```diff
@@ -139,6 +147,71 @@ void vPedalsMonitor(void* pv_params)
 		// MAX_ADC_VAL_12B) && 	!is_timer_active(&diff_timer)) 	start_timer(&diff_timer,
 		// PEDAL_FAULT_TIME); else 	cancel_timer(&diff_timer);
 
+
+		/* Evaluate accelerator faults */
+		if (is_timer_expired(&oc_timer))
+			//todo queue fault
+			continue;
+		else if ((adc_data[ACCELPIN_1] == MAX_ADC_VAL_12B || adc_data[ACCELPIN_2] == MAX_ADC_VAL_12B) &&
+			!is_timer_active(&oc_timer))
+			start_timer(&oc_timer, PEDAL_FAULT_TIME);
+		else
+			cancel_timer(&oc_timer);
+		
+		if (is_timer_expired(&sc_timer))
+			//todo queue fault
```

Comment:
Doesn't seem like you are queuing faults yet. I think all you need to do is just call `queue_fault(&fault_data);`

Reply:
added the queuing faults.